drivers/sx126x/sx126x_radio_hal.c

```diff
@@ -0,0 +1,647 @@
+            sx126x_write_buffer(dev, pos + 0x80, iol->iol_base, iol->iol_len);
+            DEBUG("[sx126x] netdev: send: wrote data to payload buffer.\n");
+            pos += iol->iol_len;
+        }
+    }
+
+    sx126x_set_lora_payload_length(dev, pos);
+    DEBUG("[sx126x] writing (size: %d).\n", (pos));
+    return 0;
+}
+
+static int _request_op(ieee802154_dev_t *hal, ieee802154_hal_op_t op, void *ctx){
+    sx126x_t *dev = hal->priv;
+    (void)dev;
+    (void)ctx;
+    switch (op) {
+        case IEEE802154_HAL_OP_TRANSMIT:
+        dev->ack_filter = false;
+        _set_state(dev, STATE_TX);
+
+        break;
+        case IEEE802154_HAL_OP_SET_RX:
+
+        _set_state(dev, STATE_RX);
+
+        break;
+
+        case IEEE802154_HAL_OP_SET_IDLE:
+
+        _set_state(dev, STATE_IDLE);
+
+        break;
+
+        case IEEE802154_HAL_OP_CCA:
+            DEBUG("[sx126x] netdev: HAL_OP_CCA (CAD Detection state)\n");
+            dev->cad_detected = 0;
+            _set_state(dev, STATE_IDLE);
+            sx126x_set_cad(dev);
+
+        break;
+
+        default:
+        assert(false);
+        break;
+
+    }
+    return 0;
+}
+
+static int _confirm_op(ieee802154_dev_t *hal, ieee802154_hal_op_t op, void *ctx){
+    sx126x_t *dev = hal->priv;
+    (void)op;
+    (void)ctx;
+    ieee802154_tx_info_t *info = ctx;
+    bool eagain = false;
+    sx126x_state_t state;
+    _get_state(dev, &state);
+
+    
+switch (op){
+    case IEEE802154_HAL_OP_TRANSMIT:
+       if (info) {
+            info->status = (dev->cad_detected) ? TX_STATUS_MEDIUM_BUSY : TX_STATUS_SUCCESS;
+        }
+    while(state == STATE_TX)
+        _get_state(dev, &state);
+        
+    break;
+
+    case IEEE802154_HAL_OP_SET_RX:
+        if(state!=STATE_RX) 
+            eagain = true;
+    break;
+
+    case IEEE802154_HAL_OP_SET_IDLE:
+        eagain = (state != STATE_IDLE);
+        break;
+
+    case IEEE802154_HAL_OP_CCA:
+            *((bool*) ctx) = !dev->cad_detected;
+    break;
+
+    default:
+        eagain = false;
+        assert(false);
+        break;
+    
+}
+
+    if (eagain) {
+        DEBUG("EAGAIN");
+        return -EAGAIN;
+    }
+
+    return 0;
+}
+
+static int _len(ieee802154_dev_t *hal){
+    (void)hal;
+    sx126x_t *dev = hal->priv;
+    sx126x_rx_buffer_status_t rx_buffer_status;
+    sx126x_get_rx_buffer_status(dev, &rx_buffer_status);
+    dev->size = rx_buffer_status.pld_len_in_bytes;
+    return dev->size;
+}
+
+static int _read(ieee802154_dev_t *hal, void *buf, size_t max_size, ieee802154_rx_info_t *info)
+{
+    (void)hal;
+    (void)buf;
+    (void)info;
+
+    sx126x_t* dev = hal->priv;
+
+        /* Getting information about last received packet */
+    netdev_lora_rx_info_t *packet_info = (void*)info;
+    sx126x_rx_buffer_status_t rx_buffer_status;
+    sx126x_pkt_status_lora_t pkt_status;
+    sx126x_get_rx_buffer_status(dev, &rx_buffer_status);
+
+    dev->size = rx_buffer_status.pld_len_in_bytes;
```

Comment:
As mentioned above, you probably don't need this `dev->size` member. Please remove it from the struct and simply declare a `size` variable.